graphql/provider.go

```diff
@@ -40,10 +68,67 @@ func graphqlConfigure(ctx context.Context, d *schema.ResourceData) (interface{},
 		GQLServerUrl:   d.Get("url").(string),
 		RequestHeaders: d.Get("headers").(map[string]interface{}),
 	}
+
+	if d.Get("login_query") != "" {
+		queryResponse, resBytes, err := queryExecute(ctx, d, config, "login_query", "login_query_variables")
+		if err != nil {
+			return nil, diag.FromErr(fmt.Errorf("unable to execute read query: %w", err))
+		}
+
+		if queryErrors := queryResponse.ProcessErrors(); queryErrors.HasError() {
+			return nil, *queryErrors
+		}
+
+		var queryResponseCty cty.Value
+		if queryResponseCty, err = gocty.ToCtyValue(string(resBytes), cty.String); err != nil {
```

Comment:
Instead of returning the entire response for it to be parsed by the user, do you think it would be valuable to have an additional property for `login_query_value_attribute` (or something along those lines) that we can use to get the value for the user? 

In other words, instead of the user having to write this: 
```hcl
# These headers will be interpolated using the result from the above login query
  authenticated_headers = {
    "Authorization" = "Bearer $${jsondecode(login_query_response).data.loginAPI.accessToken}"
  }
```

The value for authenticated headers can be computed. 

What are your thoughts on this?

Reply:
Well, I wanted to keep it generic enough to adapt to GraphQL API servers that may use authentication schemes other than OAuth 2.0 (i.e. with header name != `Authorization` and/or value format != `Bearer <token>`).

I agree it would simplify both provider code and user configuration if enforcing OAuth 2.0 authorization scheme and automatically compute the `Authorization` header with a `Bearer ${login_query_value_attribute}` value.

Reply:
Good point. Now I’m wondering how we might support more than one scheme. A lot of APIs support basic auth using API keys, for example. This use case in particular is interesting tho, as I believe you are looking to solve the need to use two separate provider instances to feed the output of a token request into the main provider, right? (This is one of many reasons why I use Pulumi now for everything 😂).

Reply:
Im admittedly torn about your current API for setting `authenticated_headers`. For the most part, other than the nicety of not storing an API key in state, one would simply use the normal provider headers to set any basic auth (as Ive done with this provider in the past). Im starting to lean toward having this feature dedicated to the Oauth Bearer token scenario for that reason, which means we can compute the authenticated headers. If we went that route, we could name these properties as `oauth_login_query`, and `oauth_login_query_variables`, which leaves room for us to add support for other schemes if the use-case arises. How do you feel about that (none of this is rhetorical, I truly want to hear your opinion, as I appreciate your contribution and feedback as an active user of this provider)

Reply:
I'm good with your proposal, I'll implement it ASAP.

Reply:
sounds good, thanks again for the contribution and helping make this provider better for the community.